tests/integration/mirrorpeer_status_test.go

```diff
@@ -0,0 +1,169 @@
+						Namespace: testSCNamespace,
+					},
+				},
+			},
+		},
+	}
+)
+
+var _ = Describe("MirrorPeer Status Tests", func() {
+	When("Source secrets have been created", func() {
+		BeforeEach(func() {
+
+			err := k8sClient.Create(context.TODO(), &managedcluster1, &client.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			err = k8sClient.Create(context.TODO(), &managedcluster2, &client.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			err = k8sClient.Create(context.TODO(), &namespace1, &client.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			err = k8sClient.Create(context.TODO(), &namespace2, &client.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			pr1 := fakeMirrorPeer.Spec.Items[0]
+			pr2 := fakeMirrorPeer.Spec.Items[1]
+
+			secretNN1 := types.NamespacedName{
+				Name:      common.CreateUniqueSecretName(pr1.ClusterName, pr1.StorageClusterRef.Namespace, pr1.StorageClusterRef.Name),
+				Namespace: pr1.ClusterName,
+			}
+
+			storageClusterNN1 := types.NamespacedName{
+				Name:      pr1.StorageClusterRef.Name,
+				Namespace: pr1.StorageClusterRef.Namespace,
+			}
+
+			sec1 := common.CreateSourceSecret(secretNN1, storageClusterNN1, []byte("SecretData1"))
+
+			secretNN2 := types.NamespacedName{
+				Name:      common.CreateUniqueSecretName(pr2.ClusterName, pr2.StorageClusterRef.Namespace, pr2.StorageClusterRef.Name),
+				Namespace: pr2.ClusterName,
+			}
+
+			storageClusterNN2 := types.NamespacedName{
+				Name:      pr2.StorageClusterRef.Name,
+				Namespace: pr2.StorageClusterRef.Namespace,
+			}
+
+			sec2 := common.CreateSourceSecret(secretNN2, storageClusterNN2, []byte("SecretData2"))
+
+			err = k8sClient.Create(context.TODO(), sec1, &client.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			err = k8sClient.Create(context.TODO(), sec2, &client.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			err = k8sClient.Create(context.TODO(), fakeMirrorPeer, &client.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+		})
+
+		AfterEach(func() {
+			err := k8sClient.Delete(context.TODO(), fakeMirrorPeer, &client.DeleteOptions{})
+			err = k8sClient.DeleteAllOf(context.TODO(), &clusterv1.ManagedCluster{}, &client.DeleteAllOfOptions{
+				ListOptions: client.ListOptions{
+					Namespace: "",
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			err = k8sClient.Delete(context.TODO(), &namespace1, &client.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			err = k8sClient.Delete(context.TODO(), &namespace2, &client.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should have Phase to be ExchangedSecret", func() {
+			Eventually(func() multiclusterv1alpha1.PhaseType {
+				var mp multiclusterv1alpha1.MirrorPeer
+				err := k8sClient.Get(context.TODO(), fakeMirrorPeerNamespacedName, &mp)
+				Expect(err).NotTo(HaveOccurred())
+				return mp.Status.Phase
+			}, 20*time.Second, 2*time.Second).Should(Equal(multiclusterv1alpha1.ExchangedSecret))
+
```

Comment:
Need cases for nil -> ExchangingSecret and ExchangingSecret -> ExchangedSecret.
Also, what happens after ExchangedSecret, if secret is deleted?

Reply:
> Need cases for nil -> ExchangingSecret and ExchangingSecret -> ExchangedSecret

I have added a case when the secrets are not yet created. MP will be in ExchangingSecret phase and the case where it is in `ExchangedSecret` phase when it secrets have been created

> what happens after ExchangedSecret, if secret is deleted

I have thought about this and this case has not been handled yet as it will required some rewrite in the secret controller. The MirrorPeer object needs revamped design to handle the multiple scenarios where secrets will be updated, deleted, created etc. Also update multiple mirrorpeers at the same time.

Current logic deals with a single mirrorpeer object and reconciles until it finds the proper secrets.

Reply:
> Need cases for nil -> ExchangingSecret and ExchangingSecret -> ExchangedSecret

I have added a case when the secrets are not yet created. MP will be in ExchangingSecret phase and the case where it is in `ExchangedSecret` phase when it secrets have been created

> what happens after ExchangedSecret, if secret is deleted

I have thought about this and this case has not been handled yet as it will required some rewrite in the secret controller. The MirrorPeer object needs revamped design to handle the multiple scenarios where secrets will be updated, deleted, created etc. Also update multiple mirrorpeers at the same time.

Current logic deals with a single mirrorpeer object and reconciles until it finds the proper secrets.